npm/rake-dist.js

```diff
@@ -0,0 +1,49 @@
+var child_process = require('child_process');
+var fs = require('fs');
+
+var Log = require('./log.js');
+var Build = require('./build.js');
+var build = new Build();
+var log = new Log();
+
+if (!fs.existsSync('build')) {
+  fs.mkdirSync('build');
+}
+
+var opalCompile = function(require, output) {
+  var stdout = child_process.execSync('opal --compile -I lib -I asciidoctor -g asciidoctor -r ' + require + ' --dynamic-require ignore --no-opal --no-exit > build/' + output);
+  process.stdout.write(stdout);
+}
+
+var opalCompileExtension = function(name) {
+  var stdout = child_process.execSync('opal --compile -I extensions-lab/lib -I asciidoctor -g asciidoctor -r ' + name + ' --dynamic-require ignore --no-opal --no-exit > build/asciidoctor-' + name + '.js');
+  process.stdout.write(stdout);
+}
+
+log.title('compile Asciidoctor libraries');
+opalCompile('asciidoctor/converter/docbook5', 'asciidoctor-docbook5.js');
+opalCompile('asciidoctor/converter/docbook45', 'asciidoctor-docbook45.js');
+opalCompile('asciidoctor/extensions', 'asciidoctor-extensions.js');
+opalCompile('asciidoctor', 'asciidoctor-core.js');
+
+build.replaceFileSync('build/asciidoctor-core.js', /'VERSION', "(\d+\.\d+.\d+)\.(.*)"/, "\1-\2");
+
+log.title('compile extensions-lab libraries');
+if (fs.existsSync('extensions-lab/lib')) {
+  opalCompileExtension('chrome-inline-macro');
+  opalCompileExtension('man-inline-macro');
+  opalCompileExtension('emoji-inline-macro');
+  opalCompileExtension('chart-block-macro');
+} else {
+  log.error("Unable to cross-compile extensions because git submodule 'extensions-lab' is not initialized.");
+  log.info("To initialize the submodule use the following command `git submodule init` and `git submodule update`.");
+  process.exit(9);
+}
+
+log.title('copy resources');
+log.debug('copy asciidoctor.css');
+// FIXME: Find a more robust implementation to get the Asciidoctor gem install path
+asciidoctorGemPath = child_process.execSync("ruby -e \"print (Gem::Specification.find_by_name 'asciidoctor').full_gem_path\"");
+asciidoctorCSSFile = asciidoctorGemPath + '/data/stylesheets/asciidoctor-default.css';
+fs.createReadStream(asciidoctorCSSFile).pipe(fs.createWriteStream('build/asciidoctor.css'));
+fs.createReadStream(asciidoctorCSSFile).pipe(fs.createWriteStream('examples/asciidoctor.css'));
```

Comment:
I don't think we need the stylesheet in the examples directory. That was only there because of the legacy Rake task. I think we can just reference the one in build.